### PR TITLE
hotfix handle base price list pricing correctly

### PR DESCRIPTION
**Pull request by teosarca**

### CODE REVIEW

**Comment by metas-rc** [comment]
https://github.com/metasfresh/metasfresh/issues/13589

File: backend/de.metas.business/src/main/java/de/metas/pricing/rules/price_list_version/PriceListVersionPricingRule.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+import lombok.NonNull;
+import org.adempiere.exceptions.AdempiereException;
+import org.compiere.model.I_M_PriceList_Version;
+
+import javax.annotation.Nullable;
+import java.time.ZonedDateTime;
+import java.util.HashSet;
+import java.util.function.Supplier;
+
+public class PriceListVersionPricingRule implements IPricingRule
```
Comment (metas-ts) [inline backend/de.metas.business/src/main/java/de/metas/pricing/rules/price_list_version/PriceListVersionPricingRule.java:44]:
```suggestion
/**
 * This pricing rule is a composite; see the constructor for the included sub-rules.
 */
public class PriceListVersionPricingRule implements IPricingRule
```

File: backend/de.metas.business/src/main/java/de/metas/pricing/rules/price_list_version/PriceListVersionPricingRule.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+		return pricingRule;
+	}
+
+	@Override
+	public boolean applies(final IPricingContext pricingCtx, final IPricingResult result)
+	{
+		return includedPricingRules.applies(pricingCtx, result);
+	}
+
+	@Override
```
Comment (metas-ts) [inline backend/de.metas.business/src/main/java/de/metas/pricing/rules/price_list_version/PriceListVersionPricingRule.java:83]:
```suggestion
	/**
	 * Iterates the ctx's {@code pricingCtx}'s priceList's PLVs from a PLV to its respective base-PLV 
	 * and applies the sub-pricing-rules until a price is found or all PLvs were tried.
	 */
	@Override
```

File: backend/de.metas.business/src/main/sql/postgresql/system/20-de.metas.business/5655120_sys_replace_PriceListVersion_pricing_rule.sql [hunk 3]
```diff
@@ -0,0 +1,5 @@
+-- 2022-09-07T14:37:36.691Z
```
Comment (metas-ts) [inline backend/de.metas.business/src/main/sql/postgresql/system/20-de.metas.business/5655120_sys_replace_PriceListVersion_pricing_rule.sql:1]: looking at `5655670_sys_PriceListVersionPricingRule_Classname.sql` i guess we can entirely drop this one, right?

**Review by metas-ts — APPROVED** [review]

File: backend/de.metas.business/src/main/sql/postgresql/system/20-de.metas.business/5655120_sys_replace_PriceListVersion_pricing_rule.sql (hunk 3 above)
Comment (teosarca) [reply backend/de.metas.business/src/main/sql/postgresql/system/20-de.metas.business/5655120_sys_replace_PriceListVersion_pricing_rule.sql]: refactored

**Pull request merged**